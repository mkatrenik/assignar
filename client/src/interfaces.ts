export enum ImgurImageSource {
  subreddit = 'subreddit',
  album = 'album'
}
