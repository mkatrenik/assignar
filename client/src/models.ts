import { ImgurRestApi } from '../../server/src/typings/imgur-rest-api'
import { urlToDataURL, getImageSize } from './utils'
import { TImageSize, TDataUrl } from './interfaces'
import { API_URL } from './api'
import { IImagesResponse } from '../../server/src/images/interfaces/api'

export abstract class GalleryItem {
  readonly id: string

  /**
   * DataUrl representation of image
   */
  dataUrl?: TDataUrl

  /**
   * struct representing image dimensions
   */
  size?: TImageSize

  /**
   * link - either url or dataurl
   */
  link?: string

  constructor(id: string) {
    this.id = id
  }

  /**
   * computes image dimensions
   */
  async setImageSize() {
    if (!this.dataUrl) {
      throw new Error(`this.dataUrl not set`)
    }
    this.size = await getImageSize(this.dataUrl)
  }

  /**
   * convert image from link to dataUrl
   */
  async setDataUrl() {}
}

export class GalleryItemImgur extends GalleryItem {
  private readonly _link: string

  constructor(item: ImgurRestApi.GalleryItem) {
    super(item.id)
    this._link = item.link.replace(/.jpg$/, `m.jpg`)
  }

  async setDataUrl() {
    this.dataUrl = await urlToDataURL(this._link)
  }

  get link() {
    return this._link
  }
}

export class GalleryItemTmp extends GalleryItem {
  dataUrl: string

  constructor(item: TDataUrl) {
    super(GalleryItemTmp.generateId())
    this.dataUrl = item
  }

  /**
   * generate fake id for local images
   */
  static generateId() {
    return Math.random()
      .toString(36)
      .replace(/[^a-z]+/g, '')
      .substr(0, 6)
  }

  get link() {
    return this.dataUrl
  }
}

export class GalleryItemLocal extends GalleryItem {
  readonly link: string

  constructor(item: IImagesResponse) {
    super(String(item.id))
    this.link = `${API_URL}${item.link}`
  }

  async setDataUrl() {
    this.dataUrl = await urlToDataURL(this.link)
  }
}
