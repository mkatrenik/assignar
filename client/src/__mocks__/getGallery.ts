export const mock = [
  {
    tags: [],
    width: 2016,
    datetime: 1518753106,
    account_url: null,
    link: 'https://i.imgur.com/TKQPXk2.jpg',
    type: 'image/jpeg',
    favorite: false,
    ups: null,
    in_most_viral: false,
    vote: null,
    animated: false,
    size: 742388,
    description: null,
    title: 'Chapel on the Rock, Colorado [2016x1512][OC]',
    comment_count: null,
    has_sound: false,
    in_gallery: false,
    favorite_count: null,
    bandwidth: 24139488208,
    nsfw: false,
    section: 'EarthPorn',
    downs: null,
    ad_url: '',
    id: 'TKQPXk2',
    account_id: null,
    ad_type: 0,
    points: null,
    height: 1512,
    is_album: false,
    views: 32516,
    is_ad: false,
    score: 30523
  },
  {
    downs: null,
    section: 'EarthPorn',
    nsfw: false,
    ad_url: '',
    id: 'q2uohuJ',
    account_id: null,
    ad_type: 0,
    height: 2808,
    points: null,
    is_ad: false,
    score: 57419,
    is_album: false,
    views: 270260,
    description: null,
    size: 906798,
    title: 'Mt Seymour, Vancouver BC Canada 3264x2448 [OC]',
    in_gallery: false,
    comment_count: null,
    has_sound: false,
    bandwidth: 245071227480,
    favorite_count: null,
    favorite: false,
    ups: null,
    animated: false,
    in_most_viral: false,
    vote: null,
    account_url: null,
    link: 'https://i.imgur.com/q2uohuJ.jpg',
    tags: [],
    width: 2808,
    datetime: 1518751204,
    type: 'image/jpeg'
  },
  {
    title: 'Cane River, Natchitoches, La. [OC] [3929 x 2947]',
    description: null,
    size: 559547,
    bandwidth: 148654291943,
    favorite_count: null,
    in_gallery: false,
    comment_count: null,
    has_sound: false,
    ad_url: '',
    id: 'pprl8Dk',
    account_id: null,
    downs: null,
    section: 'EarthPorn',
    nsfw: false,
    is_ad: false,
    score: 263717,
    views: 265669,
    is_album: false,
    ad_type: 0,
    height: 2304,
    points: null,
    account_url: null,
    link: 'https://i.imgur.com/pprl8Dk.jpg',
    tags: [],
    width: 3072,
    datetime: 1518750003,
    type: 'image/jpeg',
    favorite: false,
    animated: false,
    in_most_viral: false,
    vote: null,
    ups: null
  },
  {
    type: 'image/jpeg',
    link: 'https://i.imgur.com/KMebpmj.jpg',
    account_url: null,
    width: 3024,
    datetime: 1518742804,
    tags: [],
    animated: false,
    vote: null,
    in_most_viral: false,
    ups: null,
    favorite: false,
    bandwidth: 892871455169,
    favorite_count: null,
    in_gallery: false,
    has_sound: false,
    comment_count: null,
    title: 'Zion National Park [OC] 3024x4032',
    description: null,
    size: 4024681,
    score: 188833,
    is_ad: false,
    views: 221849,
    is_album: false,
    points: null,
    height: 4032,
    ad_type: 0,
    account_id: null,
    id: 'KMebpmj',
    ad_url: '',
    downs: null,
    nsfw: false,
    section: 'EarthPorn'
  },
  {
    title: 'Sunset, Neil Island, Andaman and Nicobar Islands,[OC] 3264x2448',
    size: 800896,
    description: null,
    favorite_count: null,
    bandwidth: 653994053888,
    comment_count: null,
    has_sound: false,
    in_gallery: false,
    id: '0XeEmtO',
    ad_url: '',
    account_id: null,
    section: 'EarthPorn',
    nsfw: false,
    downs: null,
    views: 816578,
    is_album: false,
    is_ad: false,
    score: 121768,
    ad_type: 0,
    height: 2448,
    points: null,
    tags: [],
    datetime: 1518715803,
    width: 3264,
    account_url: null,
    link: 'https://i.imgur.com/0XeEmtO.jpg',
    type: 'image/jpeg',
    favorite: false,
    in_most_viral: false,
    vote: null,
    animated: false,
    ups: null
  },
  {
    is_album: false,
    views: 23596,
    score: 19775,
    is_ad: false,
    points: null,
    height: 5304,
    ad_type: 0,
    account_id: null,
    id: 'N170Iao',
    ad_url: '',
    section: 'EarthPorn',
    nsfw: false,
    downs: null,
    favorite_count: null,
    bandwidth: 94897000636,
    has_sound: false,
    comment_count: null,
    in_gallery: false,
    title: 'Buena Vista, Colorado (OC) [7952 x 5304]',
    size: 4021741,
    description: null,
    vote: null,
    in_most_viral: false,
    animated: false,
    ups: null,
    favorite: false,
    type: 'image/jpeg',
    datetime: 1518655835,
    width: 7952,
    tags: [],
    link: 'https://i.imgur.com/N170Iao.jpg',
    account_url: null
  },
  {
    is_album: false,
    views: 425538,
    is_ad: false,
    score: 112105,
    ad_type: 0,
    height: 1080,
    points: null,
    id: 'iCtomjC',
    ad_url: '',
    account_id: null,
    nsfw: false,
    section: 'EarthPorn',
    downs: null,
    favorite_count: null,
    bandwidth: 170056899864,
    comment_count: null,
    has_sound: false,
    in_gallery: false,
    title:
      "Today is Oregon's birthday. Here is a picture of her misty mountain foothills. [OC][1620x1080]",
    size: 399628,
    description: null,
    in_most_viral: false,
    vote: null,
    animated: false,
    ups: null,
    favorite: false,
    type: 'image/jpeg',
    tags: [],
    datetime: 1518625803,
    width: 1620,
    account_url: null,
    link: 'https://i.imgur.com/iCtomjC.jpg'
  },
  {
    size: 331735,
    description: null,
    title:
      'Icelandic landscape - June 2017. Taken on a trip North of Reykjavik [OC][4032x3024]',
    has_sound: false,
    comment_count: null,
    in_gallery: false,
    favorite_count: null,
    bandwidth: 58428485550,
    section: 'EarthPorn',
    nsfw: false,
    downs: null,
    account_id: null,
    ad_url: '',
    id: 'a4G2gFu',
    points: null,
    height: 1704,
    ad_type: 0,
    is_album: false,
    views: 176130,
    score: 161456,
    is_ad: false,
    width: 2272,
    datetime: 1518603003,
    tags: [],
    link: 'https://i.imgur.com/a4G2gFu.jpg',
    account_url: null,
    type: 'image/jpeg',
    favorite: false,
    ups: null,
    vote: null,
    in_most_viral: false,
    animated: false
  },
  {
    title: 'Arambol beach, Goa. [OC] [4160X3120]',
    size: 423902,
    description: null,
    favorite_count: null,
    bandwidth: 466094661668,
    comment_count: null,
    has_sound: false,
    in_gallery: false,
    id: 'dyln4by',
    ad_url: '',
    account_id: null,
    nsfw: false,
    section: 'EarthPorn',
    downs: null,
    views: 1099534,
    is_album: false,
    is_ad: false,
    score: 468883,
    ad_type: 0,
    points: null,
    height: 3120,
    tags: [],
    datetime: 1518591603,
    width: 4160,
    account_url: null,
    link: 'https://i.imgur.com/dyln4by.jpg',
    type: 'image/jpeg',
    favorite: false,
    in_most_viral: false,
    vote: null,
    animated: false,
    ups: null
  },
  {
    type: 'image/jpeg',
    account_url: null,
    link: 'https://i.imgur.com/DUZVPZV.jpg',
    tags: [],
    datetime: 1518572403,
    width: 1920,
    animated: false,
    in_most_viral: false,
    vote: null,
    ups: null,
    favorite: false,
    bandwidth: 56212585232,
    favorite_count: null,
    in_gallery: false,
    comment_count: null,
    has_sound: false,
    title: '[OC]Avalanche Lake, Montana, USA[1920x1442]',
    description:
      'Details:\n\nShot with Panasonic GH3\n20mm Panasonic Lumix lens\nf/1.7, 1/400 sec., 125 ISO\n1920 x 1442\nSaturation, contrast, vibrancy boosts, and sharpening performed in Paintshop Pro x9\n\nEDIT: Unedited version: https://imgur.com/gallery/7d7qKp4',
    size: 693332,
    is_ad: false,
    score: 65475,
    views: 81076,
    is_album: false,
    ad_type: 0,
    height: 1442,
    points: null,
    id: 'DUZVPZV',
    ad_url: '',
    account_id: null,
    downs: null,
    section: 'EarthPorn',
    nsfw: false
  },
  {
    favorite: false,
    vote: null,
    in_most_viral: false,
    animated: false,
    ups: null,
    datetime: 1518544203,
    width: 1920,
    tags: [],
    link: 'https://i.imgur.com/ECzq95K.jpg',
    account_url: null,
    type: 'image/jpeg',
    account_id: null,
    ad_url: '',
    id: 'ECzq95K',
    nsfw: false,
    section: 'EarthPorn',
    downs: null,
    views: 48980,
    is_album: false,
    score: 40670,
    is_ad: false,
    points: null,
    height: 1080,
    ad_type: 0,
    title: 'A starry sunrise [1920x1080] [OC]',
    size: 249355,
    description: null,
    favorite_count: null,
    bandwidth: 12213407900,
    has_sound: false,
    comment_count: null,
    in_gallery: false
  }
]
