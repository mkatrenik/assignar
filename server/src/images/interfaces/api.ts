import { CreateImagePayload } from '../dto/create-image-payload'

export type TGalleryResponse = ImgurRestApi.Response<ImgurRestApi.GalleryItem[]>
export type TImageUploadResponse = ImgurRestApi.Response<ImgurRestApi.Image>
export type TUploadPayloadArgs = CreateImagePayload & { buffer: Buffer }
