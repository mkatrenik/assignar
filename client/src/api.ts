import axios from 'axios'
import { GetGalleryOptions } from '../../server/src/images/dto/get-gallery-options'
import { ImgurRestApi } from '../../server/src/typings/imgur-rest-api'
import { mock } from './__mocks__/getGallery'

axios.defaults.baseURL = 'http://localhost:3000'

export async function getGallery(options: GetGalleryOptions) {
  console.log(options)

  return (mock as any) as ImgurRestApi.GalleryItem[]
  // const res = await axios.get<ImgurRestApi.GalleryImage[]>('/api/v1/images', {
  //   params: options
  // })
  // return res.data
}
