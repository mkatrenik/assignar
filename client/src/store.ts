import { store } from 'react-easy-state'
import { ImgurRestApi } from '../../server/src/typings/imgur-rest-api'
import { getGallery } from './api'
import { GetGalleryOptions } from '../../server/src/images/dto/get-gallery-options'
import { ImgurImageSource } from './interfaces'

export class State {
  images: ImgurRestApi.GalleryItem[] = []
  imgurImageSourceType = ImgurImageSource.subreddit
  imgurImageSourceValue = ''

  async fetch(opts: GetGalleryOptions) {
    const data = await getGallery(opts)
    this.images = data
  }
}

let appState = store(new State())
;(window as any).appState = appState

export function resetAppState(state: State = new State()) {
  appState = store(state)
}

export { appState }
