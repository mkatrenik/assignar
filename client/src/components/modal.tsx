import * as React from 'react'
import { view } from 'react-easy-state'
import Modal from 'react-responsive-modal'
import { appState } from '../store'
import { MakeMosaic } from './mosaic'
import { TileRenderer } from '../interfaces'

@view
export class ModalCmp extends React.Component {
  box: HTMLDivElement | null = null

  /**
   * grab svg element and call upload
   */
  onUploadBtnClick = async () => {
    if (this.box) {
      // get svg elem from mosaic component, because there is no api for
      // getting back rendered svg
      const svg = this.box.querySelector('svg')

      if (svg) {
        await appState.upload(svg)
      } else {
        console.error(`Couldn't find svg!`)
      }
    }
  }

  render() {
    return (
      <div>
        <Modal
          open={appState.modalIsOpen}
          onClose={() => {
            appState.modalIsOpen = false
            appState.newLink = ''
          }}
          little
        >
          <div
            ref={box => (this.box = box)}
            style={{ marginTop: 30, minWidth: 100 }}
          >
            <select
              style={{ marginBottom: 10 }}
              value={appState.currentTileRenderer}
              onChange={ev =>
                (appState.currentTileRenderer = ev.currentTarget
                  .value as TileRenderer)
              }
            >
              <option value={TileRenderer.circleRenderer}>
                {TileRenderer.circleRenderer}
              </option>
              <option value={TileRenderer.halfToneRenderer}>
                {TileRenderer.halfToneRenderer}
              </option>
            </select>
            {appState.loading && <div>LOADING...</div>}
            {appState.selectedImage && (
              <>
                <MakeMosaic />
                {appState.loading === false && (
                  <div>
                    <button onClick={this.onUploadBtnClick}>Upload</button>
                  </div>
                )}
                {appState.newLink !== '' && (
                  <div>
                    <a target="_blank" href={appState.newLink}>
                      {appState.newLink}
                    </a>
                  </div>
                )}
              </>
            )}
          </div>
        </Modal>
      </div>
    )
  }
}
