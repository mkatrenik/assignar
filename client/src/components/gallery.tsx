import * as React from 'react'
import { view } from 'react-easy-state'
import { appState } from '../store'
import styled from 'styled-components'
import { ImgurImageSource } from '../interfaces'
import { ModalCmp } from './modal'
import { GalleryItemLocal } from '../models'
import { fileReaderPromised } from '../utils'

const Styled = styled.div`
  img {
    object-fit: cover;
    width: 100%;
    max-height: 200px;
  }
  .select {
    width: 100%;
  }
  .name {
    width: 100%;
  }
`

@view
export class Gallery extends React.Component {
  constructor(props: any) {
    super(props)
    this.handleFileChange = this.handleFileChange.bind(this)
  }

  async componentDidMount() {
    appState.fetch()
  }

  async handleFileChange(ev: React.ChangeEvent<HTMLInputElement>) {
    const fileList = ev.currentTarget.files

    if (!fileList) return
    if (fileList.length === 0) return

    const dataUrl = await fileReaderPromised(fileList[0])

    appState.imagesLocal.push(new GalleryItemLocal(dataUrl))
  }

  render() {
    return (
      <Styled>
        <form
          onSubmit={ev => {
            appState.imagesCurrentPage = 0
            appState.fetch()
            ev.preventDefault()
          }}
        >
          <div className="grid">
            <div className="col-1">
              <select
                value={appState.imgurImageSourceType}
                className="select"
                onChange={ev => {
                  appState.imgurImageSourceValue = ''
                  appState.imgurImageSourceType = ev.currentTarget.value as any
                }}
              >
                <option value={ImgurImageSource.subreddit}>Subreddit</option>
                <option value={ImgurImageSource.album}>Imgur album</option>
              </select>
            </div>
            <div className="col-2">
              <input
                type="text"
                className="name"
                value={appState.imgurImageSourceValue}
                placeholder={
                  appState.imgurImageSourceType === ImgurImageSource.subreddit
                    ? 'Subreddit name (e.g. EarthPorn or aww)'
                    : 'Album id (uploads album is K5Ror)'
                }
                onChange={ev =>
                  (appState.imgurImageSourceValue = ev.currentTarget.value)
                }
              />
            </div>
            <div className="col-1">
              <button
                type="submit"
                disabled={appState.imgurImageSourceValue === ''}
              >
                fetch
              </button>
            </div>
            <div className="col">
              Or upload your own{' '}
              <input
                type="file"
                name="image"
                onChange={this.handleFileChange}
                accept="image/*"
              />
            </div>
          </div>
        </form>
        <div className="local-images">
          {appState.imagesLocal.length > 0 && (
            <>
              <h2>Local images</h2>
              <div className="grid">
                {appState.imagesLocal.slice(0, 12).map(i => (
                  <img
                    key={i.id}
                    onClick={() => {
                      appState.onImageClick(i)
                    }}
                    className="col-2 image"
                    src={i.link}
                  />
                ))}
              </div>
            </>
          )}
        </div>
        <div className="imgur-images">
          <h2>Imgur images from {appState.imgurImageSourceValue}</h2>
          Page: {appState.imagesCurrentPage}{' '}
          <button
            onClick={_ => {
              appState.setImagesCurrentPage(-1)
              appState.fetch()
            }}
          >
            prev
          </button>{' '}
          <button
            onClick={_ => {
              appState.setImagesCurrentPage(1)
              appState.fetch()
            }}
          >
            next
          </button>
          <div style={{ marginBottom: 10 }} />
          <div className="grid">
            {appState.images.filter(i => i.link.match(/jpg$/)).map(i => (
              <img
                key={i.id}
                onClick={() => {
                  appState.onImageClick(i)
                }}
                className="col-2 image"
                src={i.link}
              />
            ))}
          </div>
        </div>
        {appState.modalIsOpen && <ModalCmp />}
      </Styled>
    )
  }
}
