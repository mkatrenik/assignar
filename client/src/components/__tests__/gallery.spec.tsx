import * as React from 'react'
import { shallow } from 'enzyme'
import { Gallery } from '../gallery'
import { resetAppState, appState } from '../../store'
import * as api from '../../api'
import { mock } from '../../__mocks__/getGallery'
import { ImgurImageSource } from '../../interfaces'
import { GalleryItemTmp, GalleryItemLocal } from '../../models'
import { ModalCmp } from '../modal'

describe('<Gallery />', () => {
  beforeEach(() => {
    resetAppState()
  })

  afterEach(() => jest.clearAllMocks())

  it('should show imgur images', async () => {
    appState.images = mock as any

    const wrapper = shallow(<Gallery />, { disableLifecycleMethods: true })
    expect(wrapper).toMatchSnapshot()
  })

  it('should call getImgurImages on didmount', async () => {
    const spy = jest
      .spyOn(appState, 'getImgurImages')
      .mockImplementation(() => null)

    shallow(<Gallery />)
    expect(spy.mock.calls.length).toBe(1)
  })

  it('should call getImgurImages on submit', async () => {
    const wrapper = shallow(<Gallery />, { disableLifecycleMethods: true })

    const spy = jest
      .spyOn(appState, 'getImgurImages')
      .mockImplementation(() => null)

    wrapper
      .find('select')
      .simulate('change', { currentTarget: { value: ImgurImageSource.album } })

    wrapper
      .find('.name')
      .simulate('change', { currentTarget: { value: 'foo' } })

    wrapper.find('form').simulate('submit', { preventDefault() {} })

    expect(spy.mock.calls.length).toBe(1)
    expect(appState.imgurImageSourceType).toBe(ImgurImageSource.album)
    expect(appState.imgurImageSourceValue).toBe('foo')
  })

  it('should handle local file upload from computer', async () => {
    const wrapper = shallow(<Gallery />, { disableLifecycleMethods: true })

    const fakeEvent: any = {
      currentTarget: {
        files: [new Blob()]
      }
    }
    await (wrapper.instance() as Gallery).handleFileChange(fakeEvent)
    expect(appState.selectedImage).toBeInstanceOf(GalleryItemTmp)
  })

  it('should show local images', async () => {
    appState.imagesLocal.push(
      new GalleryItemLocal({ id: 1, link: 'foo', title: 'asda' })
    )
    const wrapper = shallow(<Gallery />, { disableLifecycleMethods: true })
    expect(wrapper).toMatchSnapshot()
  })

  it('should call getLocalImages on didmount', async () => {
    const spy = jest
      .spyOn(appState, 'getLocalImages')
      .mockImplementation(() => null)

    shallow(<Gallery />)
    expect(spy.mock.calls.length).toBe(1)
  })

  it('should call appState.onImageClick on image click', async () => {
    appState.imagesLocal.push(
      new GalleryItemLocal({ id: 1, link: 'foo', title: 'asda' })
    )
    const wrapper = shallow(<Gallery />, { disableLifecycleMethods: true })
    const spy = jest.spyOn(appState, 'onImageClick').mockReturnValue('foo')

    wrapper
      .find('.local-images .image')
      .first()
      .simulate('click')

    expect(spy.mock.calls.length).toBe(1)
    expect(spy.mock.calls[0][0]).toBeInstanceOf(GalleryItemLocal)
  })

  it('should open modal dialog on appStore.modalIsOpen', async () => {
    appState.modalIsOpen = true
    const wrapper = shallow(<Gallery />, { disableLifecycleMethods: true })
    expect(wrapper.find(ModalCmp).length).toBe(1)
  })
})
