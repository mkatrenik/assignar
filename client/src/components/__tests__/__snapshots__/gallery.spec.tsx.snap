// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gallery /> should show data 1`] = `
<styled.div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="grid"
    >
      <div
        className="col-1"
      >
        <select
          className="select"
          onChange={[Function]}
          value="subreddit"
        >
          <option
            value="subreddit"
          >
            Subreddit
          </option>
          <option
            value="album"
          >
            Imgur album
          </option>
        </select>
      </div>
      <div
        className="col-2"
      >
        <input
          className="name"
          onChange={[Function]}
          placeholder="Subreddit name (e.g. EarthPorn)"
          type="text"
          value=""
        />
      </div>
      <div
        className="col-1"
      >
        <button
          type="submit"
        >
          fetch
        </button>
      </div>
      <div
        className="col"
      >
        Or upload your own
         
        <input
          accept="image/*"
          name="image"
          onChange={[Function]}
          type="file"
        />
      </div>
    </div>
  </form>
  <div
    className="local-images"
  />
  <div
    className="imgur-images"
  >
    <h2>
      Imgur images
    </h2>
    <div
      className="grid"
    >
      <img
        className="col-2 image"
        key="TKQPXk2"
        onClick={[Function]}
        src="https://i.imgur.com/TKQPXk2.jpg"
      />
      <img
        className="col-2 image"
        key="q2uohuJ"
        onClick={[Function]}
        src="https://i.imgur.com/q2uohuJ.jpg"
      />
      <img
        className="col-2 image"
        key="pprl8Dk"
        onClick={[Function]}
        src="https://i.imgur.com/pprl8Dk.jpg"
      />
      <img
        className="col-2 image"
        key="KMebpmj"
        onClick={[Function]}
        src="https://i.imgur.com/KMebpmj.jpg"
      />
      <img
        className="col-2 image"
        key="0XeEmtO"
        onClick={[Function]}
        src="https://i.imgur.com/0XeEmtO.jpg"
      />
      <img
        className="col-2 image"
        key="N170Iao"
        onClick={[Function]}
        src="https://i.imgur.com/N170Iao.jpg"
      />
      <img
        className="col-2 image"
        key="iCtomjC"
        onClick={[Function]}
        src="https://i.imgur.com/iCtomjC.jpg"
      />
      <img
        className="col-2 image"
        key="a4G2gFu"
        onClick={[Function]}
        src="https://i.imgur.com/a4G2gFu.jpg"
      />
      <img
        className="col-2 image"
        key="dyln4by"
        onClick={[Function]}
        src="https://i.imgur.com/dyln4by.jpg"
      />
      <img
        className="col-2 image"
        key="DUZVPZV"
        onClick={[Function]}
        src="https://i.imgur.com/DUZVPZV.jpg"
      />
      <img
        className="col-2 image"
        key="ECzq95K"
        onClick={[Function]}
        src="https://i.imgur.com/ECzq95K.jpg"
      />
    </div>
  </div>
</styled.div>
`;

exports[`<Gallery /> should show local images on file upload 1`] = `
<styled.div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="grid"
    >
      <div
        className="col-1"
      >
        <select
          className="select"
          onChange={[Function]}
          value="subreddit"
        >
          <option
            value="subreddit"
          >
            Subreddit
          </option>
          <option
            value="album"
          >
            Imgur album
          </option>
        </select>
      </div>
      <div
        className="col-2"
      >
        <input
          className="name"
          onChange={[Function]}
          placeholder="Subreddit name (e.g. EarthPorn)"
          type="text"
          value=""
        />
      </div>
      <div
        className="col-1"
      >
        <button
          type="submit"
        >
          fetch
        </button>
      </div>
      <div
        className="col"
      >
        Or upload your own
         
        <input
          accept="image/*"
          name="image"
          onChange={[Function]}
          type="file"
        />
      </div>
    </div>
  </form>
  <div
    className="local-images"
  >
    <React.Fragment>
      <h2>
        Local images
      </h2>
      <div
        className="grid"
      >
        <img
          className="col-2 image"
          key="foo"
          onClick={[Function]}
          src="base64"
        />
      </div>
    </React.Fragment>
  </div>
  <div
    className="imgur-images"
  >
    <h2>
      Imgur images
    </h2>
    <div
      className="grid"
    />
  </div>
</styled.div>
`;

exports[`<Gallery /> should show local images on file upload 2`] = `
<styled.div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="grid"
    >
      <div
        className="col-1"
      >
        <select
          className="select"
          onChange={[Function]}
          value="subreddit"
        >
          <option
            value="subreddit"
          >
            Subreddit
          </option>
          <option
            value="album"
          >
            Imgur album
          </option>
        </select>
      </div>
      <div
        className="col-2"
      >
        <input
          className="name"
          onChange={[Function]}
          placeholder="Subreddit name (e.g. EarthPorn)"
          type="text"
          value=""
        />
      </div>
      <div
        className="col-1"
      >
        <button
          type="submit"
        >
          fetch
        </button>
      </div>
      <div
        className="col"
      >
        Or upload your own
         
        <input
          accept="image/*"
          name="image"
          onChange={[Function]}
          type="file"
        />
      </div>
    </div>
  </form>
  <div
    className="local-images"
  >
    <React.Fragment>
      <h2>
        Local images
      </h2>
      <div
        className="grid"
      >
        <img
          className="col-2 image"
          key="foo"
          onClick={[Function]}
          src="base64"
        />
      </div>
    </React.Fragment>
  </div>
  <div
    className="imgur-images"
  >
    <h2>
      Imgur images
    </h2>
    <div
      className="grid"
    />
  </div>
</styled.div>
`;
